java/code/src/com/redhat/rhn/manager/content/ContentSyncManager.java

```diff
@@ -1168,20 +1167,34 @@ public static SUSEProduct createNewProduct(SCCProductJson p, Map<String, Channel
         return product;
     }
 
+    private List<ProductTreeEntry> loadStaticTree() {
+        String tag = Config.get().getString(ConfigDefaults.PRODUCT_TREE_TAG);
+        return loadStaticTree(tag);
+    }
+
     /*
      * load the static tree from file
      */
-    private static List<StaticInfoEntry> loadStaticTree() {
-        List<StaticInfoEntry> tree = new ArrayList<>();
+    private List<ProductTreeEntry> loadStaticTree(String tag) {
+        List<ProductTreeEntry> tree = new ArrayList<>();
         try {
-            tree = JsonParser.GSON.fromJson(new BufferedReader(new InputStreamReader(
-                            new FileInputStream(sumaProductTreeJson))),
-                    SCCClientUtils.toListType(StaticInfoEntry.class));
+            List<Credentials> credentials = filterCredentials();
+            for (Credentials c : credentials) {
+                try {
+                    SCCClient scc = getSCCClient(c);
+                    tree = scc.productTree();
+                }
+                catch (SCCClientException | URISyntaxException e) {
+                    throw new ContentSyncException(e);
+                }
+            }
         }
-        catch (IOException e) {
-            log.error(e);
+        catch (ContentSyncException e) {
+            e.printStackTrace();
         }
-        return tree;
+        return tree.stream().filter(e -> {
+            return e.getTags().isEmpty() || e.getTags().contains(tag);
```

Comment:
NitPick: Maybe replace it with Lambda expression but it is not important so can be ignored.